chatkit-core/src/main/kotlin/com/pusher/chatkit/users/api/UserSubscriptionDispatcher.kt

```diff
@@ -0,0 +1,109 @@
+package com.pusher.chatkit.users.api
+
+import com.pusher.chatkit.rooms.api.JoinedRoomApiTypeMapper
+import com.pusher.chatkit.rooms.state.DeletedRoom
+import com.pusher.chatkit.rooms.state.JoinedRoom
+import com.pusher.chatkit.rooms.state.JoinedRoomsReceived
+import com.pusher.chatkit.rooms.state.JoinedRoomsState
+import com.pusher.chatkit.rooms.state.LeftRoom
+import com.pusher.chatkit.rooms.state.UpdatedRoom
+import com.pusher.chatkit.state.ChatkitState
+import org.reduxkotlin.Dispatcher
+import org.reduxkotlin.GetState
+
+internal class UserSubscriptionDispatcher(
+    val stateGetter: GetState<ChatkitState>,
+    val joinedRoomApiTypeMapper: JoinedRoomApiTypeMapper,
+    val dispatcher: Dispatcher
+) {
+
+    internal fun onEvent(event: UserSubscriptionEvent) {
+        when (event) {
+            is UserSubscriptionEvent.InitialState -> {
+
+                val currentState: JoinedRoomsState? = stateGetter().joinedRoomsState
+                if (currentState != null) {
+                    determineRoomDifferences(currentState, event)
+                } else {
+                    dispatcher(JoinedRoomsReceived(
+                            rooms = joinedRoomApiTypeMapper.toManyRoomInternal(event.rooms),
```

Comment:
I'm not sure why ktlint is not flagging that up but that's double indent. For me IDE makes automatically (as I type or reformat) single indent. We need to resolve it as it will become more an more of an issue going forward and ktlint was supposed to stop us from having such issues. When it comes to IDE do you have Project:Project scheme selected in Settings -> Editor -> Code Style?

Reply:
I did not 😅hopefully that was the issue - I guess because I did a lot of things to try to set it up this one was easy for me to miss (and easier for others to adopt)?

Reply:
cool it's still a puzzle why ktlint is not catching [it](https://ktlint.github.io/#rule-indentation), though now it will be less of an issue if our IDEs are set up fine

Reply:
So I went digging - it looks like my IDE was still trying to override the indentation for continuation to 8 spaces - I've configured it to do only 4.